integration-tests/src/main/java/org/apache/druid/testing/DockerConfigProvider.java

```diff
@@ -260,4 +269,32 @@ public String getStreamEndpoint()
       }
     };
   }
+
+  // there is probably a better way to do this...
+  static class ArbitraryPropertiesJsonDeserializer extends JsonDeserializer<Map<String, String>>
+  {
+    @Override
+    public Map<String, String> deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
+        throws IOException
+    {
+      // reading like this results in a map that has both nested objects and also flattened string pairs
+      // so the map looks something like this:
+
+      //    {
+      //      "a" : { "b": { "c" : "d" }}},
+      //      "a.b.c":"d"
+      //    }
+
+      // filtering out the top level keys which do not have string values produces what we want here that
+      // '-Ddruid.test.config.properites.some.property.key=foo' -> { "some.property.key":"foo"}
+      Map<String, Object> parsed = jsonParser.readValueAs(Map.class);
+      Map<String, String> flat = new HashMap<>();
+      for (Map.Entry<String, Object> entry : parsed.entrySet()) {
+        if (entry.getValue() instanceof String) {
+          flat.put(entry.getKey(), (String) entry.getValue());
+        }
+      }
```

Comment:
How does this flatten the map? It looks like it's just ignoring any value that isn't a String?

Reply:
TBH I'm not entirely sure why the map ends up like it does coming from jackson, but I tried to explain what is going on here in the comments above. This block is just filtering out non-string values, leaving only the flat map after it is done.

Reply:
I misread your comment earlier I thought an input of 

```json
{"a" : { "b": { "c" : "d" }}}}
```

would be converted to a map with an entry like

```
"a.b.c":"d"
```

To help me understand, could you describe an example (sample input + expected output) where the default Deserializer would produce a different output from what this deserializer does?

Reply:
If its not been discussed before, its the `org.apache.druid.guice.JsonConfigurator#hieraricalPutValue` behind such kind of duplicate values. Its possible that passing properties such as below may work 
`druid.test.config.properties={"kafka.test.property.security.protocol": "SASL_SSL", "kafka.test.property.sasl.mechanism":"PLAIN"}`